addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1492,6 +1493,183 @@ options.registry.gallery = options.Class.extend({
+            this.$target.removeClass('show-end-message');
+        }
+        this._refreshPublicWidgets();
+    },
+    /**
+     * Changes the countdown due date.
+     *
+     * @see this.selectClass for parameters
+     */
+    endTime: function (previewMode, value, $opt) {
+        var momentValue = moment(new Date($opt.find('input').val()));
+        this.endTimeValue = momentValue.unix();
+        this.$target[0].dataset.endTime = this.endTimeValue;
+        this._refreshPublicWidgets();
+    },
+    /**
+    * Changes the countdown style.
+    *
+    * @see this.selectClass for parameters
+    */
+    layout: function (previewMode, value, $opt) {
+        switch (value) {
+            case 'circle':
+                this.$target[0].dataset.shape = 'circle';
+                this.$target[0].dataset.progressBarStyle = 'disappear';
+                this.$target[0].dataset.progressBarWeight = 'thin';
+                this.$target[0].dataset.shapeBackground = 'none';
+                break;
+            case 'boxes':
+                this.$target[0].dataset.shape = 'boxes';
+                this.$target[0].dataset.progressBarStyle = 'none';
+                this.$target[0].dataset.shapeBackground = 'plain';
+                break;
+            case 'clean':
+                this.$target[0].dataset.shape = 'none';
+                this.$target[0].dataset.progressBarStyle = 'none';
+                this.$target[0].dataset.shapeBackground = 'none';
+                break;
+        }
+        this.$target[0].dataset.layout = value;
+        this._refreshPublicWidgets();
+    },
+    /**
+    * Changes the URL the countdown redirects to once finished.
+    *
+    * @see this.selectClass for parameters
+    */
+    redirectUrl: function (previewMode, value, $opt) {
+        var val = $opt.find('input').val();
+        this.redirectUrlValue = val;
+        this.$target[0].dataset.redirectUrl = val;
+        this._refreshPublicWidgets();
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _setActive: function () {
+        this._super(...arguments);
+        this.$el.find('[data-display]').removeClass('active')
+            .filter('[data-display=' + this.$target[0].dataset.display + ']').addClass('active');
+        this.$el.find('[data-end-action]').removeClass('active')
+            .filter('[data-end-action=' + this.$target[0].dataset.endAction + ']').addClass('active');
+        this.$el.find('[data-layout]').removeClass('active')
+            .filter('[data-layout=' + this.$target[0].dataset.layout + ']').addClass('active');
+
+        this.$el.find('[data-end-time] > input').datetimepicker('date', moment(this.endTimeValue * 1000));
+        this.$el.find('[data-redirect-url] > input').val(this.redirectUrlValue);
+    },
```

Comment:
You left unneeded things since you now use `select-data-attribute`

Reply:
I already removed all the unnecessary ones (forgot `display`).

I need to keep `end-action` and `layout` still they use their own data-method and not just select data attribute.
- `end-action` is removing and/or adding the end message into to the dom & some other stuff
- `layout` is actually a preset of other dataset (that can be modified each in `advanced` mode). It is updating all other dataset element if you change the preset.
Eg `circle` is `shape = 'circle'`, `progressBarStyle = 'disappear'`, `progressBarWeight = 'thin'`, `shapeBackground = 'none'` but those are hidden unless `advanced` mode is chosen.